### PR TITLE
⭐ add userIdGroupPairs and prefixListIds to secGroup IpPerms

### PR DESCRIPTION
**Pull request by vjeffrey**
and a bit of cleanup too

Screenshot 2024-07-08 at 19 07 26

https://github.com/mondoohq/cnquery/issues/4322

### CODE REVIEW

File: providers/aws/resources/aws.lr.manifest.yaml [hunk 1]
```diff
@@ -1219,7 +1219,11 @@ resources:
       ipProtocol: {}
       ipRanges: {}
       ipv6Ranges: {}
+      prefixListIds:
+        min_mondoo_version: latest
```
Comment (tas50) [inline providers/aws/resources/aws.lr.manifest.yaml]:
```suggestion
        min_mondoo_version: 9.0.0
```

File: providers/aws/resources/aws.lr.manifest.yaml [hunk 2]
```diff
@@ -1219,7 +1219,11 @@ resources:
...
       toPort: {}
+      userIdGroupPairs:
+        min_mondoo_version: latest
```
Comment (tas50) [inline providers/aws/resources/aws.lr.manifest.yaml]:
```suggestion
        min_mondoo_version: 9.0.0
```

File: providers/aws/resources/aws.lr [hunk 3]
```diff
@@ -2973,6 +2973,10 @@ private aws.ec2.securitygroup.ippermission @defaults("id toPort fromPort ipProto
   ipRanges []string
   // IPv6 ranges
   ipv6Ranges []string
+  // List of Prefix ids
```
Comment (tas50) [inline providers/aws/resources/aws.lr]:
```suggestion
  // List of Prefix IDs
```

File: providers/aws/resources/aws.lr [hunk 4]
```diff
@@ -2973,6 +2973,10 @@ private aws.ec2.securitygroup.ippermission @defaults("id toPort fromPort ipProto
...
+  prefixListIds []dict
+  // List of user id group pairs
```
Comment (tas50) [inline providers/aws/resources/aws.lr]:
```suggestion
  // List of user ID group pairs
```

File: providers/aws/resources/aws_ec2.go [hunk 5]
```diff
@@ -400,104 +408,25 @@ func (a *mqlAwsEc2) getSecurityGroups(conn *connection.AwsConnection) []*jobpool
...
-					if err != nil {
-						return nil, err
-					}
-					res = append(res, mqlS3SecurityGroup)
+					res = append(res, mqlEc2SecurityGroup)
+					mqlEc2SecurityGroup.(*mqlAwsEc2Securitygroup).cacheIpPerms = group.IpPermissions
+					mqlEc2SecurityGroup.(*mqlAwsEc2Securitygroup).cacheIpPermsEgress = group.IpPermissionsEgress
+					mqlEc2SecurityGroup.(*mqlAwsEc2Securitygroup).groupId = *group.GroupId
+					mqlEc2SecurityGroup.(*mqlAwsEc2Securitygroup).region = region
+					mqlEc2SecurityGroup.(*mqlAwsEc2Securitygroup).cacheVpc = group.VpcId
```
Comment (preslavgerchev) [inline providers/aws/resources/aws_ec2.go:429]: does that persist across query runs? this updates the MQL object after it's been stored

**Review by scottford-io — COMMENTED** [review]
@vjeffrey thank you for this. I am able to get the config needed to properly test AWS Security groups ...

image

File: providers/aws/resources/aws_ec2.go (hunk 5 above)
Comment (vjeffrey) [reply providers/aws/resources/aws_ec2.go:429]: ya, I believe so. we use this same pattern a lot in the aws code

**Review by scottford-io — APPROVED** [review]
@vjeffrey thank you for this. Looks good!

**Pull request merged**